code/game/gamemodes/devil/devilinfo.dm

```diff
@@ -0,0 +1,440 @@
+
+/proc/randomdevilbanish()
+	return pick(BANISH_WATER, BANISH_COFFIN, BANISH_FORMALDYHIDE, BANISH_RUNES, BANISH_CANDLES, BANISH_DESTRUCTION, BANISH_FUNERAL_GARB)
+
+/datum/devilinfo/proc/add_soul(datum/mind/soul)
+	if(soulsOwned.Find(soul))
+		return
+	soulsOwned += soul
+	owner.current.nutrition = NUTRITION_LEVEL_FULL
+	owner.current << "<span class='warning'>You feel satiated as you received a new soul.</span>"
+	update_hud()
+	switch(SOULVALUE)
+		if(0)
+			owner.current << "<span class='warning'>Your hellish powers have been restored."
+			give_base_spells()
+		if(BLOOD_THRESHOLD)
+			increase_blood_lizard()
+		if(TRUE_THRESHOLD)
+			increase_true_devil()
+		if(ARCH_THRESHOLD)
+			increase_arch_devil()
+
+/datum/devilinfo/proc/remove_soul(datum/mind/soul)
+	if(soulsOwned.Remove(soul))
+		check_regression()
+		owner.current << "<span class='warning'>You feel as though a soul has slipped from your grasp.</span>"
+		update_hud()
+
+/datum/devilinfo/proc/check_regression()
+	if(form == ARCH_DEVIL)
+		return //arch devil can't regress
+	//Yes, fallthrough behavior is intended, so I can't use a switch statement.
+	if(form == TRUE_DEVIL && SOULVALUE < TRUE_THRESHOLD)
+		regress_blood_lizard()
+	if(form == BLOOD_LIZARD && SOULVALUE < BLOOD_THRESHOLD)
+		regress_humanoid()
+	if(SOULVALUE < 0)
+		remove_spells()
+		owner.current << "<span class='warning'>As punishment for your failures, all of your powers except contract creation have been revoked."
+
+/datum/devilinfo/proc/increase_form()
+	switch(form)
+		if(BASIC_DEVIL)
+			increase_blood_lizard()
+		if(BLOOD_LIZARD)
+			increase_true_devil()
+		if(TRUE_DEVIL)
+			increase_arch_devil()
+
+/datum/devilinfo/proc/regress_humanoid()
+	owner.current << "<span class='warning'>Your powers weaken, have more contracts be signed to regain power."
+	if(istype(owner.current, /mob/living/carbon/human))
+		var/mob/living/carbon/human/H = owner.current
+		H.set_species(/datum/species/human, 1)
+		H.regenerate_icons()
+	give_base_spells()
+	if(istype(owner.current.loc, /obj/effect/dummy/slaughter/))
+		owner.current.forceMove(get_turf(owner.current))//Fixes dying while jaunted leaving you permajaunted.
+	form = BASIC_DEVIL
+
+/datum/devilinfo/proc/regress_blood_lizard()
+	var/mob/living/carbon/true_devil/D = owner.current
+	D << "<span class='warning'>Your powers weaken, have more contracts be signed to regain power."
+	D.oldform.loc = D.loc
+	owner.transfer_to(D.oldform)
+	give_lizard_spells()
+	qdel(D)
+	form = BLOOD_LIZARD
+	update_hud()
+
+
+/datum/devilinfo/proc/increase_blood_lizard()
+	owner.current << "<span class='warning'>You feel as though your humanoid form is about to shed.  You will soon turn into a blood lizard."
+	sleep(50)
+	if(istype(owner.current, /mob/living/carbon/human))
+		var/mob/living/carbon/human/H = owner.current
+		H.set_species(/datum/species/lizard, 1)
+		H.underwear = "Nude"
+		H.undershirt = "Nude"
+		H.socks = "Nude"
+		H.dna.features["mcolor"] = "511" //A deep red
```

Comment:
Yeah, there will have to be a lot of testing, but before I can even get to that point, I need to cleanup the compile errors.  I'm about halfway done.